pandas_gbq/schema.py

```diff
@@ -42,8 +49,99 @@ def _clean_schema_fields(fields):
     return clean_schema
 
 
+def schema_difference(schema_remote, schema_local):
+    """Calculates schemas difference and formats the output.
+
+    Parameters
+    ----------
+    schema_remote : dict
+        Schema for comparison. Each item of ``fields`` should have a 'name'
+        and a 'type'
+    schema_local : dict
+        Schema for comparison. Each item of ``fields`` should have a 'name'
+        and a 'type'
+
+    Returns
+    -------
+    str
+        Formatted schema difference output
+    """
+    _schema_difference = _calculate_schema_difference(
+        schema_remote, schema_local
+    )
+    return _format_schema_difference(_schema_difference)
+
+
+def _format_schema_difference(schema_difference):
+    """Formats the schema difference.
+
+    By default it shows only 3 differences. In case of more - it
+    says how many more are left.
+
+    Parameters
+    ----------
+    schema_difference : list[str]
+        List of differences between schemas.
+
+    Returns
+    -------
+    str
+        Formatted schema difference output
+    """
+    if len(schema_difference) < 4:
+        diff_to_show = "\n".join(schema_difference)
+    else:
+        diffs_left = len(schema_difference) - 3
+        schema_difference = schema_difference[:3]
+        if diffs_left != 0:
+            schema_difference.append("And {} more left.".format(diffs_left))
+        diff_to_show = "\n".join(schema_difference)
+    return diff_to_show
+
+
+def _calculate_schema_difference(schema_remote, schema_local):
+    """Calculates difference in dataframe and BigQuery schemas.
+
+    Compares dataframe and BigQuery schemas to identify exact differences
+    in each field (field can be missing in the dataframe or field can have
+    a different type).
+
+    Parameters
+    ----------
+    schema_remote : dict
+        Schema for comparison. Each item of ``fields`` should have a 'name'
+        and a 'type'
+    schema_local : dict
+        Schema for comparison. Each item of ``fields`` should have a 'name'
+        and a 'type'
+
+    Returns
+    -------
+    List[str]
+        List of field differences
+    """
+    fields_remote = _clean_schema_fields(schema_remote.get("fields", []))
+    fields_local = _clean_schema_fields(schema_local.get("fields", []))
+    diff = []
+    for field_remote in fields_remote:
+        for field_local in fields_local:
+            if field_local["name"] == field_remote["name"]:
+                if field_local["type"] != field_remote["type"]:
```

Comment:
This might miss type mismatches if the order of the columns is different in either `field_remote` or `field_local` (or does `_clean_schema_fields` sort them?). I think we'd want:

1. Loop through only `fields_local` (dataframe)
2. Check if the field name isn't in `fields_remote` (table)
3. Check if the field types don't match